collections/CHANGELOG.md

```diff
@@ -31,6 +31,10 @@ Ref: https://keepachangelog.com/en/1.0.0/
 
 ## [Unreleased]
 
+### Features
+
+* [#16074](https://github.com/cosmos/cosmos-sdk/pull/16607) - Introduces `Clear` method for `Map` and `KeySet`
```

Comment:
Shouldn't we split the collections changes, tag them and release a new collection with this feature instead of using replaces? This was people on v0.50 and any other SDK chains can directly get this feature.
Right now, we need to add replaces, and it becomes quickly unclear where to release collections.

Reply:
I can cherrypick the collection changes into a separate PR, I'm just not sure if I want to mark the `Clear` API as stable (yet), this is why it ended up in a replace block.

Reply:
This makes sense too, then we can use a pseudo version here as well. Imho it is better than adding replaces for such packages and it simplifies the release process as well.

Reply:
@julienrbrt unfortunately the pseudo-version does not work well because go will think "0.2.0" > "pseudo-version" and will not upgrade it for the modules.

Reply:
Really, the pseudo version that Go gets me with latest commit on main is `v0.2.1-0.20230620040119-e078f1a49e8b` so it should work.

```sh
$ go list -f '{{.Version}}' -m cosmossdk.io/collections@main
v0.2.1-0.20230620040119-e078f1a49e8b
```

Reply:
I cherrypicked this PR's commits here: https://github.com/cosmos/cosmos-sdk/pull/16618

After that is merged I will update the go.mods using collections@main.